slam-libraries/viam-cartographer/src/slam_service/config.cc

```diff
@@ -47,6 +50,26 @@ int ParseAndValidateConfigParams(int argc, char** argv) {
     LOG(INFO) << "data_rate_ms: " << FLAGS_data_rate_ms << "\n";
     LOG(INFO) << "map_rate_sec: " << FLAGS_map_rate_sec << "\n";
 
+    slamService.data_dir = FLAGS_data_dir;
+    slamService.config_params = FLAGS_config_param;
+    slamService.port = FLAGS_port;
+    slamService.sensors = FLAGS_sensors;
+    slamService.data_rate_ms = std::chrono::milliseconds(FLAGS_data_rate_ms);
+    slamService.map_rate_sec = std::chrono::seconds(FLAGS_map_rate_sec);
+
+    slamService.slam_mode =
+        ConfigParamParser(slamService.config_params, "mode=");
+    if (slamService.slam_mode.empty()) {
+        LOG(ERROR) << "slam mode is missing\n";
```

Comment:
Can we add unit tests for the failures in this function? Maybe in https://viam.atlassian.net/browse/DATA-577?

Reply:
Good idea, will add a comment to the ticket.